scrapy/downloadermiddlewares/httpcache.py

```diff
@@ -11,41 +13,54 @@
     TimeoutError,
 )
 from twisted.web.client import ResponseFailed
+from typing import Optional, Tuple, Type, Union
```

Comment:
`typing` is a standard module so it should be in the first import block :)

Reply:
Agreed. Change is made.